lib/tape.js

```diff
@@ -0,0 +1,259 @@
+/**
+ * Codemod for transforming Tape tests into AVA.
+ */
+
+const tapeToAvaAsserts = {
+	fail: 'fail',
+	pass: 'pass',
+
+	ok: 'truthy',
+	true: 'truthy',
+	assert: 'truthy',
+
+	notOk: 'falsy',
+	false: 'falsy',
+	notok: 'falsy',
+
+	error: 'ifError',
+	ifErr: 'ifError',
+	iferror: 'ifError',
+
+	equal: 'is',
+	equals: 'is',
+	isEqual: 'is',
+	strictEqual: 'is',
+	strictEquals: 'is',
+
+	notEqual: 'not',
+	notStrictEqual: 'not',
+	notStrictEquals: 'not',
+	isNotEqual: 'not',
+	doesNotEqual: 'not',
+	isInequal: 'not',
+
+	deepEqual: 'deepEqual',
+	isEquivalent: 'deepEqual',
+	same: 'deepEqual',
+
+	notDeepEqual: 'notDeepEqual',
+	notEquivalent: 'notDeepEqual',
+	notDeeply: 'notDeepEqual',
+	notSame: 'notDeepEqual',
+	isNotDeepEqual: 'notDeepEqual',
+	isNotEquivalent: 'notDeepEqual',
+	isInequivalent: 'notDeepEqual',
+
+	skip: 'skip',
+	throws: 'throws',
+	doesNotThrow: 'notThrows'
+};
+
+const unsupportedTestFunctions = new Set([
+	// No equivalent in AVA:
+	'timeoutAfter',
+	'deepLooseEqual',
+	'looseEqual',
+	'looseEquals',
+	'notDeepLooseEqual',
+	'notLooseEqual',
+	'notLooseEquals'
+]);
+
+/**
+ * Updates CommonJS and import statements from Tape to AVA
+ * @return string with test function name if transformations were made
+ */
+function updateTapeRequireAndImport(j, ast) {
+	let testFunctionName = null;
+	ast.find(j.CallExpression, {
+		callee: {name: 'require'},
+		arguments: arg => arg[0].value === 'tape'
+	})
+	.filter(p => p.value.arguments.length === 1)
+	.forEach(p => {
+		p.node.arguments[0].value = 'ava';
+		testFunctionName = p.parentPath.value.id.name;
+	});
+
+	ast.find(j.ImportDeclaration, {
+		source: {
+			type: 'Literal',
+			value: 'tape'
+		}
+	})
+	.forEach(p => {
+		p.node.source.value = 'ava';
+		testFunctionName = p.value.specifiers[0].local.name;
+	});
+
+	return testFunctionName;
+}
+
+export default function tapeToAva(fileInfo, api) {
+	const j = api.jscodeshift;
+	const ast = j(fileInfo.source);
+
+	const testFunctionName = updateTapeRequireAndImport(j, ast);
+
+	if (testFunctionName) {
```

Comment:
Stylistic: Maybe just `return` right here, rather than indenting all of the transform? For an `if` block this long, this requires a later maintainer to scroll all the way to the bottom of the file to check for any logic that might exist when `!testFunctionName`.